### DISCUSSION TITLE
RP Pico Ejects Itself When Being Written To

### DISCUSSION THREAD

**Discussion by conordgould** [Moddable-OpenSource/moddable#1179] [Q&A]
Hello! I have three RP-Pico W's and have tried building the WiFi  example to all three, with the same "Disk not ejected properly" notification. My terminal outputs this, which I do not understand:

```
/dev/cu.usbmodem14201
Setting programming modeproduct/vendor match: /dev/cu.usbmodem14201
Waiting for /Volumes/RPI-RP2...
Copying: /Users/cgould/Documents/Development/SDKs/moddable/build/bin/pico/debug/WeatherLink/xs_pico.elf to RPI-RP2
Connect to xsbug @ 2e8a:000a.
product/vendor match: /dev/cu.usbmodem14201
```

And I get this notification every single time:

<img width="347" alt="image" src="https://github.com/Moddable-OpenSource/moddable/assets/73129917/289e2548-87a4-4bd9-b4cf-2a9ba88c5534">

The pico does not get written to at all.

**Comment by phoddie** [comment]
I see you've closed this, so perhaps you've already resolved the problem.

The "Disk Not Ejected Properly" message is a not actually an error in this case. The notification is displayed after the firmware image is copied and the Pico is rebooted. The reboot causes the volume to go off-line which looks to macOS like you unplugged it. So, macOS displays a warning. This [blog post](https://blog.adafruit.com/2021/05/11/how-to-tone-down-macos-big-surs-circuitpy-eject-notifications/) explains how to suppress the warning.
